lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp

```diff
@@ -167,56 +149,122 @@ TEST_F(MmeAppProcedureTest, TestImsiAttachEpsOnlyDetach) {
   EXPECT_CALL(*spgw_handler, sgw_handle_delete_session_request()).Times(1);
 
   // Construction and sending Initial Attach Request to mme_app mimicing S1AP
-  message_p = itti_alloc_new_message(TASK_S1AP, S1AP_INITIAL_UE_MESSAGE);
-
   uint8_t nas_msg[]       = {0x07, 0x41, 0x71, 0x08, 0x09, 0x10, 0x10, 0x00,
                        0x00, 0x00, 0x00, 0x10, 0x02, 0xe0, 0xe0, 0x00,
                        0x04, 0x02, 0x01, 0xd0, 0x11, 0x40, 0x08, 0x04,
                        0x02, 0x60, 0x04, 0x00, 0x02, 0x1c, 0x00};
   uint32_t nas_msg_length = 31;
+  send_mme_app_initial_ue_msg(&nas_msg[0], nas_msg_length, plmn);
+
+  // Sending AIA to mme_app mimicing successful S6A response for AIR
+  send_authentication_info_resp(imsi);
+
+  // Constructing and sending Authentication Response to mme_app mimicing S1AP
+  uint8_t nas_msg2[] = {0x07, 0x53, 0x10, 0x66, 0xff, 0x47, 0x2d,
+                        0xd4, 0x93, 0xf1, 0x5a, 0x00, 0x00, 0x00,
+                        0x00, 0x00, 0x00, 0x00, 0x00};
+  nas_msg_length     = 19;
+  send_mme_app_uplink_data_ind(&nas_msg2[0], nas_msg_length, plmn);
+
+  // Constructing and sending SMC Response to mme_app mimicing S1AP
+  uint8_t nas_msg3[] = {0x47, 0xc0, 0xb5, 0x35, 0x6b, 0x00, 0x07,
+                        0x5e, 0x23, 0x09, 0x33, 0x08, 0x45, 0x86,
+                        0x34, 0x12, 0x31, 0x71, 0xf2};
+  nas_msg_length     = 19;
+  send_mme_app_uplink_data_ind(&nas_msg3[0], nas_msg_length, plmn);
```

Comment:
We use these same exact messages in two places, IMSI and GUTI attach tests. Is there value in defining these in a macro?